libraries/AP_Common/AP_Common.h

```diff
@@ -99,47 +99,14 @@
 
 // @}
 
-
-////////////////////////////////////////////////////////////////////////////////
-/// @name	Types
-///
-/// Data structures and types used throughout the libraries and applications. 0 = default
-/// bit 0: Altitude is stored               0: Absolute,	1: Relative
-/// bit 1: Change Alt between WP            0: Gradually,	1: ASAP
-/// bit 2: Direction of loiter command      0: Clockwise	1: Counter-Clockwise
-/// bit 3: Req.to hit WP.alt to continue    0: No,          1: Yes
-/// bit 4: Relative to Home					0: No,          1: Yes
-/// bit 5: Loiter crosstrack reference      0: WP center    1: Tangent exit point
-/// bit 6:
-/// bit 7: Move to next Command             0: YES,         1: Loiter until commanded
-
-//@{
-
-struct PACKED Location_Option_Flags {
-    uint8_t relative_alt : 1;           // 1 if altitude is relative to home
-    uint8_t unused1      : 1;           // unused flag (defined so that loiter_ccw uses the correct bit)
-    uint8_t loiter_ccw   : 1;           // 0 if clockwise, 1 if counter clockwise
-    uint8_t terrain_alt  : 1;           // this altitude is above terrain
-    uint8_t origin_alt   : 1;           // this altitude is above ekf origin
-    uint8_t loiter_xtrack : 1;          // 0 to crosstrack from center of waypoint, 1 to crosstrack from tangent exit location
+template<typename s, int s_size, int t> struct _assert_storage_size {
```

Comment:
EWWWWW! Can we just do a normal assert where we need it? This is far more confusing...

Reply:
As discussed on the call, there's merit in being able to see what's changed.

I've added the following:
```
// assert_storage_size template: assert that the memory used to store an
// item is of a specific size.
// example invocation:
// assert_storage_size<class Location, 16> _assert_storage_size_Location;
// templates are used for this because the compiler's output will
// usually contain details of the template instantiation so you can
// see how the actual size differs from the expected size.
```